analyses/sample-distribution-analysis/01-filter-across-types.R

```diff
@@ -0,0 +1,162 @@
+  # Given the name of a primary site, create a vector containing the disease 
+  # types expressed within the primary site.
+  # 
+  # Note: the disease types found at all instances of the location substring 
+  #       will be included 
+  #
+  # Args:
+  #   location: the name of a primary site found within the dataset
+  #
+  # Returns:
+  #   disease_type_vector: the vector of disease types expressed at the 
+  #                        named primary site
+  disease_type_vector <- brain_location %>%
+    dplyr::arrange(dplyr::desc(n)) %>%
+    dplyr::filter(stringr::str_detect(primary_site, location)) %>%
+    dplyr::pull(disease_type_new)
+  unique(disease_type_vector)
+}
+
+# Define the file.path to output directories
+outputDir <- file.path("analyses", "sample-distribution-analysis")
+
+# Create directories to hold the output.
+if (!dir.exists(file.path(outputDir, "results"))) {
+  dir.create(file.path(outputDir, "results"))
+}
+if (!dir.exists(file.path(outputDir, "plots"))) {
+  dir.create(file.path(outputDir, "plots"))
+}
+
+# Read in dataset
+df2 <- data.frame(readr::read_tsv(
+  file.path("data", "pbta-histologies.tsv")
+))
+
+# Remove na's
+df2 <- df2 %>%
+  dplyr::filter(!is.na(disease_type_new))
+
+# data.frame with the count of each unique cancer type expression
+disease_expression <- df2 %>%
+  dplyr::group_by(disease_type_new) %>%
+  dplyr::count(name = "count") %>%
+  dplyr::arrange(dplyr::desc(count))
+
+# Calculate the total count of the dataset
+sum_count <- sum(disease_expression$count)
+
+# Create a percent variable and round to 4 decimal places 
+# (so values will have 2 decimal places as percentages) 
+disease_expression <- disease_expression %>%
+  dplyr::mutate(percent = paste0((round(count / sum_count, 4) * 100), "%"))
+
+# Reorder the columns to be displayed in descending order by count on the plot
+disease_expression$disease_type_new <- with(disease_expression,
+                                            reorder(disease_type_new, -count))
+
+# Write to tsv file
+readr::write_tsv(disease_expression,
+                 file.path(outputDir, "results",
+                           "disease_expression.tsv"))
+
+# Create a bar plot of sample distribution across cancer types
+gg_types <- disease_expression %>%
+  ggplot2::ggplot(ggplot2::aes(x = disease_type_new, y = count, fill = count)) +
+  ggplot2::geom_col() +
+  ggplot2::theme_bw() +
+  ggplot2::labs(x = "Cancer Types", y = "Count",
+                title = "Sample Distribution Across Cancer Types") +
+  ggplot2::scale_y_continuous(breaks = seq(0, 500, by = 100)) +
+  ggplot2::theme(axis.text.x = ggplot2::element_text(
+    angle = 75,
+    hjust = 1,
+    size = 8
+  ),
+  panel.grid = ggplot2::element_blank()) +
+  ggplot2::geom_text(nudge_y = 6.5, size = 2,
+                     ggplot2::aes(label = paste0(disease_expression$percent)))
+
+# Save plot
+ggplot2::ggsave(
+  gg_types,
+  file = file.path(outputDir, "plots", "distribution_across_cancer_types.pdf"),
```

Comment:
Could then change this to
```suggestion
  file = file.path(plots_dir, "distribution_across_cancer_types.pdf"),
```